lib/src/generateInit.cc

```diff
@@ -107,6 +110,16 @@ void genInit(const NNmodel &model,          //!< Model description
     std::string oB = "", cB = "";
 #endif // _WIN32
 
+    // **NOTE** if we are using GCC on x86_64, bugs in some version of glibc can cause bad performance issues.
+    // Best solution involves setting LD_BIND_NOW=1 so check whether this has been applied
+    os << "#if defined(__GNUG__) && !defined(__clang__) && defined(__x86_64__) && __GLIBC__ == 2 && (__GLIBC_MINOR__ == 23 || __GLIBC_MINOR__ == 24)" << std::endl;
+    os << "    const char *ldBindNow = std::getenv(\"LD_BIND_NOW\");" << std::endl;
+    os << "    if(ldBindNow == NULL || strcmp(ldBindNow, \"1\") != 0) {" << std::endl;
```

Comment:
A minor point: while I think it's customary to set `LD_BIND_NOW` to 1, from `ld`'s point of view it only has to be present, i.e. any value is fine.

Reply:
Excellent point! Do you have any thoughts on how safe ``LD_BIND_NOW=1`` is to set on a user's environment without their explicit say so?

Reply:
You mean permanently instead of just for the run of GeNN? I think it could lead to a noticeable slowdown of program starts (never really tried this myself) because all libraries are loaded at the beginning instead of when needed. Here's a quote from the [Gentoo wiki](https://wiki.gentoo.org/wiki/Hardened/Position_Independent_Code_internals#Using_prelink_and_LD_BIND_NOW) (not sure whether this still applies on modern "fast" machines):
> But you should be careful and know that using LD_BIND_NOW is not recommended on machines where responsiveness is an issue, clicking on an icon in KDE or GNOME and waiting 20 seconds for evolution to start is sometimes inacceptable by users. In doubt, use prelink! 

Using `prelink` could actually be another option (never used it, though), but I don't think it is installed by default, so not ideal either...

Reply:
That was my fear - it wouldn't be great if genn-buildmodel set the environment variable and then the user wonders why everything launched from that terminal runs really slow